### PR TITLE
Issue #13345: Enable HideUtilityClassConstructorCheckExamplesTest

### PR DESCRIPTION
**Pull request by andreiyusupau**
Issue: #13345 
Enabled HideUtilityClassConstructorCheckExamplesTest.
Replaced its Example.txt with Example.java.
Added expected violations to test.

### CODE REVIEW

**Comment by romani** [comment]
Before pushing to GitHub, please make sure `mvn verify ` on your local pass.

**Comment by andreiyusupau** [comment]
@romani How such errors should be fixed? The comment of violation should match violation description, so it cannot be shortened. Moving comment to another line didn't help.
`[ERROR] [checkstyle] [ERROR] \checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\design\hideutilityclassconstructor\Example1.java:12: Line is longer than 85 characters (found 96). [LineLength]`
The line itself:
`class Example1 { // violation, 'Utility classes should not have a public or default constructor'`

And by the way is there some kind of predefined IDE settings (Intellij IDEA) for custom codestyle used in project? Haven't find one

**Comment by romani** [comment]
`class Example1 { // violation, 'should not have a public or default constructor'`

Message is regexp, so numerous ways to shorten it.

All that we have is https://checkstyle.org/beginning_development.html#Starting_Development

You can move violation line above and below. Search in project `// violation above` there are numerous examples

**Review by nrmancuso — CHANGES_REQUESTED** [review]
@andreiyusupau looks like we lost track of this PR, apologies. Do not be afraid to ping maintainers for review after a few days.

Why did we go from one example to four?

**Comment by andreiyusupau** [comment]
> @andreiyusupau looks like we lost track of this PR, apologies. Do not be afraid to ping maintainers for review after a few days.
> 
> Why did we go from one example to four?

Short answer - because current example didn't compile. 
Long - because it is not possible to have static fields in non-public classes (and it's impossible to have multiple public classes in one file).

**Comment by romani** [comment]
GitHub, generate website

**Comment by nrmancuso** [comment]
> Short answer - because current example didn't compile.

Yes, this is expected in most cases.

> Long - because it is not possible to have static fields in non-public classes (and it's impossible to have multiple public classes in one file).

None of the classes in the original example were public, nor are the examples in this PR. 

We don't need multiple examples here.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/hideutilityclassconstructor/Example1.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.design.hideutilityclassconstructor;
+
+// xdoc section -- start
+// violation below, 'Utility classes should not have a public or default constructor'
+class Example1 {
+
+  public Example1() {
+  }
```
[the remark this answers is not in the document]
Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/hideutilityclassconstructor/Example1.java:16]: https://github.com/checkstyle/checkstyle/pull/14174#issuecomment-1924337405

[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/hideutilityclassconstructor/Example1.java:16]: @andreiyusupau, please share javac error that you run into

**Comment by romani** [comment]
Already done in other PR https://github.com/checkstyle/checkstyle/blob/master/src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheckExamplesTest.java

**Pull request closed without merging**